### PR TITLE
[SPARK-7748] [MLLIB] Graduate spark.ml from alpha

### PR DESCRIPTION
**Pull request by mengxr**
With descent coverage of feature transformers, algorithms, and model tuning support, it is time to graduate `spark.ml` from alpha. This PR changes all `AlphaComponent` annotations to either `DeveloperApi` or `Experimental`, depending on whether we expect a class/method to be used by end users (who use the pipeline API to assemble/tune their ML pipelines but not to create new pipeline components.) `UnaryTransformer` becomes a `DeveloperApi` in this PR.

@jkbradley @harsha2010

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/ml/feature/StringIndexer.scala [hunk 1]
```diff
@@ -18,7 +18,7 @@
 package org.apache.spark.ml.feature
 
 import org.apache.spark.SparkException
-import org.apache.spark.annotation.AlphaComponent
+import org.apache.spark.annotation.{Experimental, AlphaComponent}
```
Comment (harsha2010) [inline mllib/src/main/scala/org/apache/spark/ml/feature/StringIndexer.scala]: minor point order of imports exchanged

File: mllib/src/main/scala/org/apache/spark/ml/feature/Tokenizer.scala [hunk 2]
```diff
@@ -17,19 +17,19 @@
 
 package org.apache.spark.ml.feature
 
-import org.apache.spark.annotation.AlphaComponent
+import org.apache.spark.annotation.{Experimental, AlphaComponent}
```
Comment (harsha2010) [inline mllib/src/main/scala/org/apache/spark/ml/feature/Tokenizer.scala]: same here

File: mllib/src/main/scala/org/apache/spark/ml/param/params.scala [hunk 3]
```diff
@@ -24,11 +24,11 @@ import scala.annotation.varargs
 import scala.collection.mutable
 import scala.collection.JavaConverters._
 
-import org.apache.spark.annotation.{DeveloperApi, AlphaComponent}
+import org.apache.spark.annotation.{Experimental, DeveloperApi, AlphaComponent}
```
Comment (harsha2010) [inline mllib/src/main/scala/org/apache/spark/ml/param/params.scala]: same here

File: mllib/src/main/scala/org/apache/spark/ml/regression/GBTRegressor.scala [hunk 4]
```diff
@@ -20,7 +20,7 @@ package org.apache.spark.ml.regression
 import com.github.fommil.netlib.BLAS.{getInstance => blas}
 
 import org.apache.spark.Logging
-import org.apache.spark.annotation.AlphaComponent
+import org.apache.spark.annotation.{Experimental, AlphaComponent}
```
Comment (harsha2010) [inline mllib/src/main/scala/org/apache/spark/ml/regression/GBTRegressor.scala]: same here

File: mllib/src/main/scala/org/apache/spark/ml/regression/RandomForestRegressor.scala [hunk 5]
```diff
@@ -17,7 +17,7 @@
 
 package org.apache.spark.ml.regression
 
-import org.apache.spark.annotation.AlphaComponent
+import org.apache.spark.annotation.{Experimental, AlphaComponent}
```
Comment (harsha2010) [inline mllib/src/main/scala/org/apache/spark/ml/regression/RandomForestRegressor.scala]: ditto

**Comment by harsha2010** [comment]
@mengxr +1, great! i just had very minor comments, nice to graduate from alpha, the PR looks good.

**Comment by jkbradley** [comment]
LGTM.  Confirmed no spurious uses of "alpha" remain.  Did not check for unnecessary imports.

**Comment by mengxr** [comment]
Merged into master and branch-1.4.

**Pull request closed without merging**